internal/docs/component.go

```diff
@@ -261,7 +261,8 @@ func (c *ComponentSpec) createConfigs(root string, fullConfigExample interface{}
 
 	var err error
 	if len(c.Fields) > 0 && !isEmptyRootArray {
-		advancedConfig, err := c.Fields.ConfigAdvanced(fullConfigExample)
+		var advancedConfig interface{}
+		advancedConfig, err = c.Fields.ConfigAdvanced(fullConfigExample)
```

Comment:
I see we're `panic`-ing below if this error ends up being set, which makes me wonder if having panics in the code is desirable. I did a grep and saw more of them in a few other places:

<details>
  <summary>grep -rE 'panic\(' . | grep -v '_test'</summary>

```
./cmd/tools/benthos_docs_gen/main.go:		panic(err)
./cmd/tools/benthos_docs_gen/main.go:		panic(fmt.Sprintf("Failed to generate docs for '%v': sanitised config wrong type: %T", spec.Name, confSanit))
./cmd/tools/benthos_docs_gen/main.go:		panic(fmt.Sprintf("Failed to generate docs for '%v': %v", spec.Name, err))
./cmd/tools/benthos_docs_gen/main.go:			panic(fmt.Sprintf("Failed to generate docs for '%v': %v", k, err))
./cmd/tools/benthos_docs_gen/main.go:			panic(fmt.Sprintf("Failed to generate docs for '%v': %v", k, err))
./cmd/tools/benthos_docs_gen/main.go:			panic(fmt.Sprintf("Failed to generate docs for '%v': %v", k, err))
./cmd/tools/benthos_docs_gen/main.go:			panic(fmt.Sprintf("Failed to generate docs for '%v': %v", k, err))
./cmd/tools/benthos_docs_gen/main.go:			panic(fmt.Sprintf("Failed to generate docs for '%v': %v", k, err))
./cmd/tools/benthos_docs_gen/main.go:			panic(fmt.Sprintf("Failed to generate docs for '%v': %v", k, err))
./cmd/tools/benthos_docs_gen/main.go:			panic(fmt.Sprintf("Failed to generate docs for '%v': %v", k, err))
./cmd/tools/benthos_docs_gen/main.go:			panic(fmt.Sprintf("Failed to generate docs for '%v': %v", k, err))
./cmd/tools/benthos_docs_gen/main.go:			panic(fmt.Sprintf("Failed to generate docs for '%v': %v", k, err))
./cmd/tools/benthos_docs_gen/main.go:		panic(fmt.Sprintf("Failed to generate docs for bloblang functions: %v", err))
./cmd/tools/benthos_docs_gen/main.go:		panic(fmt.Sprintf("Failed to generate docs for bloblang methods: %v", err))
./cmd/tools/benthos_config_gen/main.go:		panic(err)
./cmd/tools/benthos_config_gen/main.go:		panic(err)
./cmd/tools/benthos_config_gen/main.go:		panic(err)
./cmd/tools/benthos_config_gen/main.go:		panic(err)
./cmd/tools/benthos_config_gen/main.go:		panic(err)
./cmd/tools/benthos_config_gen/main.go:			panic("Environment values at path root")
./cmd/tools/benthos_config_gen/main.go:			panic(err)
./cmd/tools/benthos_config_gen/main.go:			panic(err)
./cmd/tools/benthos_config_gen/main.go:			panic(err)
./cmd/tools/benthos_config_gen/main.go:			panic(err)
./internal/bloblang/query/deprecated_functions.go:				panic(err)
./internal/bloblang/query/method_ctor.go:		panic(fmt.Sprintf("Conflicting method name: %v", spec.Name))
./internal/bloblang/query/functions.go:			panic(err)
./internal/bloblang/query/function_ctor.go:		panic(fmt.Sprintf("Conflicting function name: %v", spec.Name))
./internal/docs/field.go:		panic("cannot combine annotated and non-annotated options for a field")
./internal/docs/field.go:		panic("annotated field options must each have a summary")
./internal/docs/field.go:		panic("cannot combine annotated and non-annotated options for a field")
./internal/docs/component.go:		panic(err)
./internal/docs/component.go:			panic(err)
./internal/docs/component.go:			panic(err)
./internal/docs/component.go:		panic(err)
./lib/util/checkpoint/type.go:		panic(err)
./lib/util/checkpoint/type.go:		panic(err)
./lib/util/text/env_vars.go:		panic(err)
./lib/util/text/env_vars.go:		panic(err)
./lib/util/text/function_vars.go:		panic(err)
./lib/util/text/function_vars.go:		panic(err)
./lib/util/text/function_vars.go:			panic(err)
./lib/input/kafka_parts.go:			panic(err)
./lib/service/list.go:				panic(err)
```

</details>

Reply:
Generally I'd say panic should always be avoided, but in most of these cases it's within functionality used exclusively by the docs generator rather than the real benthos service, where I'd say it's not ideal but I'm not losing sleep over it.

However, if we start failing useful lints because of these then I'd be happy to put the effort into eliminating them for the sake of guarding the benthos service codebase.

Reply:
Oh, good point! I’ll let you know if any of them will trigger the linters. I was mostly concerned about the ones in bloblang.